main.js

```diff
@@ -37,19 +40,56 @@ function hide(element) {
 
 function clickSave(e) {
   e.preventDefault();
-  if (inputTitle.value === '' || inputBody.value === ''){
+  if (inputTitle.value === '' || inputBody.value === '') {
     show(errorMessage);
   } else {
     var inputIdea = new Idea(inputTitle.value, inputBody.value);
     ideas.push(inputIdea);
     inputTitle.value = '';
-    inputBody.value= '';
-   
-    var emptyHTML = '';
-    for (var i= 0; i < ideas.length; i++) {
-      emptyHTML += `<article class='card'>
-      <div class='card-top-bar'>
-        <input type='image' src='assets/star-active.svg' id='button-star' />
+    inputBody.value = '';
+    updateCardView();
+  }
+}
+
+function validateError() {
+  if (inputTitle.value && inputBody.value) {
+    hide(errorMessage);
+  }
+}
+
+function toDelete(e) {
+  if (e.target.id === 'button-delete') {
+    var targetID = e.target.parentNode.parentNode.id;
+    for (var i = 0; i < ideas.length; i++) {
+      if (targetID === ideas[i].id.toString()) {
+        ideas.splice(i, 1);
+        updateCardView();
+        break;
+      }
+    }
+  }
+}
+
+function toStar(e) {
+  if (e.target.id === 'button-star') {
+    var targetID = e.target.parentNode.parentNode.id;
+    for (var i = 0; i < ideas.length; i++) {
+      if (targetID === ideas[i].id.toString()) {
+        ideas[i].isStarred = !ideas[i].isStarred;
+        updateCardView();
+      }
+    }
+  }
+}
+
+function updateCardView() {
+  var emptyHTML = '';
```

Comment:
Is this necessary?

Reply:
No, it isn't. We'll fix that in iteration 4 pull request.